docs/input-kafka.asciidoc

```diff
@@ -42,6 +42,10 @@ This input supports connecting to Kafka over:
 
 By default security is disabled but can be turned on as needed.
 
+Due to the limitation of https://javadoc.io/doc/org.apache.kafka/kafka-clients/2.5.1/org/apache/kafka/clients/consumer/ConsumerConfig.html[Kafka client],
+this plugin does not support using a proxy when communicating to Kafka broker.
+This plugin only supports using a proxy when communicating to the Schema Registry, see <<plugins-{type}s-{plugin}-schema_registry_proxy>>
```

Comment:
```suggestion
[NOTE]
=======
This plugin does not support using a proxy when communicating to the Kafka broker.  

This plugin does support using a proxy when communicating to the Schema Registry using the <<plugins-{type}s-{plugin}-schema_registry_proxy>> option.
=======
```

Reply:
Additional thoughts for discussion (@roaksoax @kaisecheng): 

- @kaisecheng: The NOTE format you used originally is correct. I'm proposing some advanced formatting for this particular situation. The takeaway: The formatting I'm proposing here isn't something we want to adopt except in special situations. 
- The link to the kafka docs doesn't appear to have anything specific to proxy support.  Unless we can point to something more specific or explain the reason we're linking to the kafka docs, I propose that we don't include the link. There are already other links to the kafka client in the docs and it's important not to use external links unless they are highly relevant. 
- I'm hesitant to say that another vendor has limitations in their product. Let's discuss the reasons.

Reply:
The NOTE format suggested above will render like this: 
<img width="804" alt="Screen Shot 2021-10-29 at 6 38 18 PM" src="https://user-images.githubusercontent.com/35154725/139508658-36669e35-3754-4666-9b68-e7f13bd958c6.png">

Reply:
REMINDER: Earlier comments indicated that we might need to include this content for output-kafka. Have we determined if that is or isn't the needed?

Reply:
@karenzone I am with you hesitating to link the java doc to show Kafka client has a limitation on supporting proxy. 
I cannot find any doc on Kafka side explicitly saying they don't support it, but we found from the code level that they aren't. I feel more comfortable removing the link.

Reply:
Good point! I should update the output doc as well 👍